src/Test.EndToEnd.Baselining/Program.cs

```diff
@@ -0,0 +1,123 @@
+    ///   Expected_Debug\                  [Each run will create an 'enriched' version of the Expected logs also]
+    ///     
+    /// 
+    ///  Execution Steps
+    ///  ===============
+    ///  Running tests will find each series folder in Input and:
+    ///    - Load the initial baseline
+    ///    - Load each log in order
+    ///    - Replace the GUID with a 'RID' (LogIndex + ' ' + ResultIndex)
+    ///    - Baseline the current log to the Baseline-so-far
+    ///    - Write detail report lines about how each Result was baselined
+    ///    - Save the output as the Baseline-so-far
+    ///    - Compare the output to the same file in 'Expected\', if present
+    ///    - Write 'enriched' logs to Output_Debug\ and Expected_Debug\ for human investigation.
+    ///  
+    ///    
+    ///  Usage
+    ///  =====
+    ///   Get Test Content:
+    ///   git clone https://github.com/microsoft/sarif-sdk-test-content
+    ///   
+    ///   Run a Full Pass:
+    ///   "Test.EndToEnd.Baselining run C:\Code\sarif-sdk-test-content\BaselineE2E"
+    ///   
+    ///   Debug matching for Result 004 in log 001 in series "Spam\v1\WorkItem\mseng":
+    ///   To quickly debug a Result, copy and paste the first line from the Series log for the Series path and the RID (ex: 001 004) from the Result to debug.
+    ///   "Test.EndToEnd.Baselining debug C:\Code\sarif-sdk-test-content\BaselineE2E Spam\v1\WorkItem\mseng 001 004"
+    ///   
+    ///   Re-generate debug logs with the details as coded in BaseliningDetailEnricher:
+    ///   "Test.EndToEnd.Baselining rebuild-debug-logs C:\Code\sarif-sdk-test-content\BaselineE2E"
+    /// </summary>
+    class Program
+    {
+        static int Main(string[] args)
+        {
+            try
+            {
+                Program program = new Program();
+
+                return Parser.Default.ParseArguments<RunOptions, DebugOptions, RebuildDebugLogsOptions>(args).MapResult(
+                    (RunOptions options) => Program.Run(options),
+                    (DebugOptions options) => Program.Debug(options),
+                    (RebuildDebugLogsOptions options) => Program.CreateDebugLogs(options),
+                    errs => 1);
+            }
+            catch (Exception ex) when (!Debugger.IsAttached)
+            {
+                Console.WriteLine(ex.ToString());
+                return 2;
+            }
+        }
+
+        static int Run(RunOptions options)
+        {
+            BaseliningTester tester = new BaseliningTester();
+            BaseliningSummary overallSummary = tester.RunAll(options.TestRootPath);
+            Console.WriteLine($"{overallSummary}");
+
+            return 0;
+        }
+
+        static int Debug(DebugOptions options)
+        {
+            BaseliningTester tester = new BaseliningTester();
+            tester.RunSeries(Path.Combine(options.TestRootPath, BaseliningTester.InputFolderName, options.DebugSeriesPath), options.DebugLogIndex, options.DebugResultIndex);
```

Comment:
>Path.Combine(options.TestRootPath, BaseliningTester.InputFolderName, options.DebugSeriesPath [](start = 29, length = 92)

Design nit: IMO the `Program` doesn't deserve to understand what to do with the options -- in this case, combining them with information that actually belongs to the `BaseliningTester` to form a path. Why not just pass the whole options object to the `RunXxx` method (here and in the other two similar places), and let the tester deal with it? It has the additional virtue of not requiring you to modify the `RunXxx` method's signature to change when you add a new option.

Reply:
I understand your point. I wanted the classes to take independent inputs (full paths which don't require any adjustment) but I don't want the human to have to type out full paths repeatedly. This makes my Options class and Program the "frontier" between the two goals, so I translate here.